vector/src/main/java/im/vector/fragments/VectorSettingsNotificationsTroubleshootFragment.kt

```diff
@@ -0,0 +1,160 @@
+                (mBottomView as? ViewGroup)?.let {
+                    TransitionManager.beginDelayedTransition(it)
+                }
+                when (it.diagStatus) {
+                    TroubleshootTest.TestStatus.NOT_STARTED -> {
+                        mSummaryDescription.text = ""
+                        mSummaryButton.visibility = View.GONE
+                        mRunButton.visibility = View.VISIBLE
+                    }
+                    TroubleshootTest.TestStatus.RUNNING -> {
+                        mSummaryDescription.text = getString(R.string.settings_troubleshoot_diagnostic_running_status, it.currentTestIndex, it.testList.size)
+                        mSummaryButton.visibility = View.GONE
+                        mRunButton.visibility = View.GONE
+                    }
+                    TroubleshootTest.TestStatus.FAILED -> {
+
+                        //check if there are quick fixes
+                        var hasQuickFix = false
+                        testManager?.testList?.let {
+                            for (test in it) {
+                                if (test.status == TroubleshootTest.TestStatus.FAILED && test.quickFix != null) {
+                                    hasQuickFix = true
+                                    break
+                                }
+                            }
+                        }
+
+                        if (hasQuickFix) {
+                            mSummaryDescription.text = getString(R.string.settings_troubleshoot_diagnostic_failure_status_with_quickfix)
+                        } else {
+                            mSummaryDescription.text = getString(R.string.settings_troubleshoot_diagnostic_failure_status_no_quickfix)
+                        }
+                        mSummaryButton.visibility = View.VISIBLE
+                        mRunButton.visibility = View.VISIBLE
+                    }
+                    TroubleshootTest.TestStatus.SUCCESS -> {
+                        mSummaryDescription.text = getString(R.string.settings_troubleshoot_diagnostic_success_status)
+                        mSummaryButton.visibility = View.VISIBLE
+                        mRunButton.visibility = View.VISIBLE
+                    }
+                }
+            }
+
+        }
+
+        mRecyclerView.adapter = testManager?.adapter
+        testManager?.runDiagnostic()
+
+
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (requestCode == NotificationTroubleshootTestManager.REQ_CODE_APK_FIX) {
+            testManager?.retry()
+            return
+        }
+        super.onActivityResult(requestCode, resultCode, data)
+    }
+
+    override fun onDetach() {
+        testManager?.cancel()
+        super.onDetach()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        (activity as? MXCActionBarActivity)?.supportActionBar?.setTitle(R.string.settings_notification_troubleshoot)
```

Comment:
look like the title of this screen is set twice then

Reply:
title is only set here now, and not anymore in backstack listener, as fragments are always full screen